Sources/RediStack/Configuration.swift

```diff
@@ -109,11 +114,36 @@ extension RedisConnection {
             }
 
             self.address = address
+            self.username = username
             self.password = password
             self.initialDatabase = initialDatabase
             self.defaultLogger = defaultLogger ?? Configuration.defaultLogger
         }
 
+        /// Creates a new connection configuration with the provided details.
+        /// - Parameters:
+        ///     - `address`: The socket address information to use for creating the Redis connection.
+        ///     - `password`: The optional password to authenticate the connection with. The default is `nil`.
+        ///     - `initialDatabase`: The optional database index to initially connect to. The default is `nil`.
+        ///       Redis by default opens connections against index `0`, so only set this value if the desired default is not `0`.
+        ///     - `defaultLogger`: The optional prototype logger to use as the default logger instance when generating logs from the connection.
+        ///     If one is not provided, one will be generated. See `RedisLogging.baseConnectionLogger`.
+        /// - Throws: `RedisConnection.Configuration.ValidationError` if invalid arguments are provided.
+        public init(
```

Comment:
Actually, isn't this the same scenario? Also two cases where a `public init(address: ...)` would be ambiguous?